From fbb361b7084b798339a9cc4f0130071360fe9591 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Tue, 23 Apr 2019 19:28:44 +0530
Subject: [PATCH 184/767] octeontx2-pf: Check IFF_MULTICAST too when enabling
 multicast

commit 354ac4ead385482b1c844273f5b1e96abbc0633f from
git@git.assembla.com:cavium/WindRiver.linux.git

When setting multicast mode IFF_MULTICAST should also be checked
along with IFF_ALLMULTI flag.

Change-Id: I8ab9b21bb001fa3c44a86ea772abddc9645fd291
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/7989
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
index a963d832e35f..719dafb22b6b 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
@@ -1787,7 +1787,7 @@ static void otx2_set_rx_mode(struct net_device *netdev)
 	/* We don't support MAC address filtering yet */
 	if (netdev->flags & IFF_PROMISC)
 		req->mode |= NIX_RX_MODE_PROMISC;
-	else if (netdev->flags & IFF_ALLMULTI)
+	else if (netdev->flags & (IFF_ALLMULTI | IFF_MULTICAST))
 		req->mode |= NIX_RX_MODE_ALLMULTI;
 
 	otx2_sync_mbox_msg_busy_poll(&pf->mbox);
-- 
2.31.1

